src/all/comicskingdom/src/eu/kanade/tachiyomi/extension/all/comicskingdom/ComicsKingdom.kt

```diff
@@ -0,0 +1,310 @@
+package eu.kanade.tachiyomi.extension.all.comicskingdom
+
+import android.app.Application
+import android.content.SharedPreferences
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import uy.kohesive.injekt.injectLazy
+import java.text.SimpleDateFormat
+import java.util.Locale
+import kotlin.math.ceil
+import kotlin.math.roundToInt
+
+class ComicsKingdom(override val lang: String) : ConfigurableSource,
+    ParsedHttpSource() {
+
+    override val name = "Comics Kingdom"
+    override val baseUrl = "https://wp.comicskingdom.com"
```

Comment:
save baseurl to the main site domain `https://comicskingdom.com` and return requests with this domain in request methods (mangadetails, chapterlist, pagelist etc)

also override `getMangaUrl` and `getChapterUrl` for correct webview/share urls

Reply:
The api is only available on the sub domain and non api request are auto resolved to the main site

Reply:
sure